agent/agent.go

```diff
@@ -306,7 +306,11 @@ func NewAgent(options ...Option) (*Agent, error) {
 		agent.logMetadata()
 	}
 
-	agent.recorder = NewSpanRecorder(agent)
+	flushFrequency := nonTestingModeFrequency
+	if agent.testingMode {
+		flushFrequency = testingModeFrequency
+	}
+	agent.recorder = NewSpanRecorder(agent, flushFrequency)
```

Comment:
Let's just store it on the agent as `agent.flushFrequency`, so we keep this simple.

Reply:
done